core/lib/libtomcrypt/x25519.c

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2022, Technology Innovation Institute (TII)
+ */
+
+#include <crypto/crypto.h>
+#include <stdlib.h>
+#include <string.h>
+#include <tee_api_types.h>
+#include <trace.h>
+#include <utee_defines.h>
+
+#include "acipher_helpers.h"
+
+/* X25519 key is an octet string of 32 bytes */
+#define X25519_KEY_SIZE_BYTES UL(32)
+
+TEE_Result crypto_acipher_alloc_x25519_keypair(struct x25519_keypair *key,
+					       size_t key_size)
+{
+
+	size_t key_size_bytes = (size_t)(key_size / 8);
+
+	if (key == NULL)
+			return TEE_ERROR_OUT_OF_MEMORY;
+
+	memset(key, 0, sizeof(*key));
+
+	if (key_size_bytes != X25519_KEY_SIZE_BYTES)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	key->priv = calloc(1, key_size_bytes);
+	key->pub = calloc(1, key_size_bytes);
+
+	if (key->priv == NULL || key->pub == NULL)
+			return TEE_ERROR_OUT_OF_MEMORY;
+
+	return TEE_SUCCESS;
+}
+
+TEE_Result crypto_acipher_gen_x25519_key(struct x25519_keypair *key,
+					 size_t key_size)
+{
+	curve25519_key ltc_tmp_key = { };
+	size_t key_size_bytes = (size_t)(key_size / 8);
+	prng_state dummy;
```

Comment:
`= { };`

Reply:
@etienne-lms Thanks for the review! One comment on this one, the struct `prng_state` here is passed since the libtomcrypt X25519 implementation expects a valid object due to this check `LTC_ARGCHK(prng != NULL)` (https://github.com/OP-TEE/optee_os/blob/master/core/lib/libtomcrypt/src/pk/x25519/x25519_make_key.c#L30), in reality it is never used as OP-TEE has its own `prng_crypto_desc` ("prng_crypto") and all the `prng`functions declare `prng_state` as `__unused`. 

Also I am not sure if it is safe to initialize `prng_state` this way since the initialization needs to properly handle the mutex. 

Making this change produces this error during key generation:
```
D/LD:  ldelf:168 ELF (cb3e5ba0-adf1-11e0-998b-0002a5d5c51b) at 0x40014000
F/TC:? 0 trace_syscall:151 syscall #8 (syscall_check_access_rights)
F/TC:? 0 trace_syscall:151 syscall #27 (syscall_cryp_obj_alloc)
F/TC:? 0 trace_syscall:151 syscall #54 (syscall_obj_generate_key)
E/TC:1   Dead canary at end of 'stack_abt[3]' (0xe1a01fc)
E/TC:1   Panic at core/kernel/thread.c:124 <thread_check_canaries>
E/TC:1   TEE load address @ 0xe100000
E/TC:1   Call stack:
E/TC:1    0x0e108900
E/TC:1    0x0e113ec8
E/TC:1    0x0e116e58
E/TC:1    0x0e106a5c
```

One option is to remove this check `LTC_ARGCHK(prng != NULL)` from libtomcrypt implementation, but I avoid making any changes to the library.
Also note, this check is not present in other asymmetric crypto key generation functions, so this case is only valid for `X25519`.

Reply:
I believe removing the `LTC_ARGCHK(prng != NULL)` check is the best option. I don't know how large `prng_state` is, but I'm afraid it can become quite large in certain configurations.

Reply:
Thanks for the comment, `LTC_ARGCHK(prng != NULL)` removed